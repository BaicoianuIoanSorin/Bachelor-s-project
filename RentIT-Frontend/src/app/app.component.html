<nb-layout>
  <nb-layout-header subheader>
    <a href="#" (click)="toggleSidebar()">
      <nb-icon *ngIf="sidebarVisible"
               [icon]="getIconsEnum().FORWARD_ARROW_IOS">
      </nb-icon>
      <nb-icon *ngIf="!sidebarVisible"
               [icon]="getIconsEnum().FORWARD_ARROW_IOS">
      </nb-icon>
      <i class="nb-menu"></i>
    </a>
  </nb-layout-header>

  <nb-sidebar [responsive]="true">
    <div>
      <a href="">
        <img *ngIf="sidebarVisible" src="assets/logo.svg" alt="Image" width="100%" height="auto"/>
        <img *ngIf="!sidebarVisible" src="assets/favicon.svg" alt="Image" width="48px" height="auto"/>
      </a>
      <nb-menu tag="menu" [items]="menuItems"></nb-menu>
    </div>
  </nb-sidebar>

  <nb-layout-column class="colored-column-basic">
    <router-outlet></router-outlet>
  </nb-layout-column>
</nb-layout>
