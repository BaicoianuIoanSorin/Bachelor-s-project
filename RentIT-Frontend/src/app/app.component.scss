body {
  overflow: hidden;
  margin: 0;
  max-height: 85vh;
}

.nb-layout-header {
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1000;
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 75px;
  background-color: inherit;

  @media only screen and (max-width: 768px) {
    height: 60px;
  }

  @media only screen and (max-width: 472px) {
    height: 50px;
  }

}

nb-icon {
  width: 5%;
  cursor:pointer;
}

.left-section,
.center-section,
.right-section {
  flex: 1;
  display: flex;
  gap: 10px;

  @media only screen and (min-width: 769px) {
    gap: 20px;
  }
}

.center-section {
  justify-content: center;
}

.right-section {
  display: flex;
  align-items: center;
  justify-content: flex-start;

  .logo-image {
    max-width: 80px;
    height: auto;
    cursor:pointer;
  }

  @media only screen and (min-width: 473px) {
    justify-content: flex-end;
  }
}

.colored-column-basic {
  padding-top: 75px;
  margin-top: 0;
  height: calc(100vh - 75px);

  @media only screen and (max-width: 768px) {
    padding-top: 60px;
    height: calc(100vh - 60px);
  }

  @media only screen and (max-width: 472px) {
    padding-top: 50px;
    height: calc(100vh - 50px);
  }
}

nb-sidebar {
  top: 50px;
  height: 92vh;

  @media only screen and (min-width: 473px) and (max-width: 768px) {
    top: 60px;
  }

  @media only screen and (min-width: 769px) {
    top: 75px;
  }
}

.sidebar-container {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 85vh;
}
.menu-section {
  display: flex;
  flex-direction: column;
  gap: 10px;
}

.general-menu-items {
  order: 1;
}

.user-menu-items {
  margin-top: auto;
  order: 2;
}
