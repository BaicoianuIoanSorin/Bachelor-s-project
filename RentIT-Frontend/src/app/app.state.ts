import {ProductsState, ProductsStateModel} from "src/app/products/products.state";
import {Action, Selector, State, StateContext} from "@ngxs/store";
import {Injectable} from "@angular/core";
import {NbSidebarService, NbToastrService} from "@nebular/theme";
import {produce} from "immer";
import { ProductsService } from "src/api/products.service";

export class ToggleNavigationBar {
  static readonly type = '[App] Toggle Navigation bar';
  constructor() {
  }
}

export interface AppStateModel {
  sidebarVisible: boolean;
}

export const defaultsState: AppStateModel = {
  sidebarVisible: true,
}

@State<AppStateModel>({
  name: 'productsPage',
  defaults: defaultsState,
})

@Injectable()
export class AppState {
  constructor(
    private toastrService: NbToastrService,
    private productsService: ProductsService,
    private sidebarService: NbSidebarService,
  ) {
  }
}

export class AppSelector {
}
