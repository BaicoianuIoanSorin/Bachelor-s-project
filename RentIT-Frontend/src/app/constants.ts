import { NbMenuItem } from '@nebular/theme';

export enum ICONS {
  CUBE_OUTLINE = 'cube-outline',
  FORWARD_ARROW_IOS = 'arrow-ios-forward-outline',
  FORWARD_BACK_IOS = 'arrow-ios-back-outline',
  SEARCH = 'search',
  ALERT_CIRCLE_OUTLINE = 'alert-circle-outline',
  MAP_OUTLINE = 'map-outline',
  LIST_OUTLINE = 'list-outline',
  ARROW_IOS_DOWNWARD_OUTLINE = 'arrow-ios-downward-outline',
  AT_OUTLINE = 'at-outline',
  EYE_OUTLINE = 'eye-outline',
  EYE_OFF_OUTLINE = 'eye-off-2-outline',
}

export enum GENERAL_MENU_ITEM_URLS {
  PRODUCTS = '/products',
  AUTHENTICATION = '/authentication',
}

export enum BREADCRUMB_KEYS {
  PRODUCTS = 'Products',
}

export const GENERAL_MENU_ITEMS: NbMenuItem[] = [
  {
    title: BREADCRUMB_KEYS.PRODUCTS,
    icon: ICONS.CUBE_OUTLINE,
    link: GENERAL_MENU_ITEM_URLS.PRODUCTS,
  },
];

export const isEmail = (email: string): boolean => {
    let regexp = new RegExp('/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/');
    return regexp.test(email);
}
