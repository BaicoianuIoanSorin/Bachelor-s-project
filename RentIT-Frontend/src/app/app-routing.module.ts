import {NgModule} from '@angular/core';
import {ExtraOptions, RouterModule, Routes} from '@angular/router';
import {NotFoundComponent} from "src/app/not-found/not-found.component";
import {GENERAL_MENU_ITEM_URLS} from "src/app/constants";

export const routes: Routes = [
  {
    path: GENERAL_MENU_ITEM_URLS.PRODUCTS.substring(1),
    loadChildren: () => import('./products/products.module').then(m => m.ProductsModule)
  },
  {
    path: '',
    redirectTo: GENERAL_MENU_ITEM_URLS.PRODUCTS,
    pathMatch: 'full',
  },
  {
    path: '**',
    title: "Oops! Couldn't find this page!",
    component: NotFoundComponent,
  }
];

const routerOptions: ExtraOptions = {
  onSameUrlNavigation: 'reload'
};

@NgModule({
  imports: [RouterModule.forRoot(routes, routerOptions)],
  exports: [RouterModule]
})
export class AppRoutingModule {
}
