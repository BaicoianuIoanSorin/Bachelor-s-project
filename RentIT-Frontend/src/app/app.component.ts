import {Component, Inject, OnInit} from '@angular/core';
import {NB_WINDOW, NbMenuItem, NbMenuService, NbSidebarService} from "@nebular/theme";
import {GENERAL_MENU_ITEMS, ICONS, NOT_LOGGED_IN_CONTEXT_MENU} from "src/app/constants";
import {filter, map} from "rxjs/operators";
import {Router} from "@angular/router";
import {Select, Store} from "@ngxs/store";
import {ProductsSelector} from "src/app/products/products.selector";
import {Observable} from "rxjs";
import {AppSelector} from "src/app/app.state";
import {UserService} from "src/api/user.service";

@Component({
  selector: 'app-root',
  template: `
    <nb-layout>
      <nb-layout-header>
        <div class="left-section">
          <nb-icon [icon]="ICONS.LIST_OUTLINE" (click)="toggleSidebar()"></nb-icon>
          <i class="nb-menu"></i>
        </div>

        <div class="center-section">
          <search-bar [icon]="ICONS.SEARCH"></search-bar>
        </div>

        <div class="right-section">
          <a href="">
            <img [src]="getImageBySize()" alt="Image" width="100%" height="auto"/>
          </a>
          <nb-user *ngIf="!(isFetching$ | async)" name="Nikita Poltoratsky"
                   title="full-stack developer"
                   [nbContextMenu]="getContextMenu()"
                   nbContextMenuTag="my-context-menu">
          </nb-user>
        </div>
      </nb-layout-header>


      <nb-sidebar [responsive]="true">
        <div>
          <nb-menu tag="menu" [items]="sidebarMenu$ | async"></nb-menu>
        </div>
      </nb-sidebar>

      <nb-layout-column class="colored-column-basic">
        <router-outlet></router-outlet>
      </nb-layout-column>
    </nb-layout>`,
  styleUrls: ['./app.component.scss']
})
export class AppComponent implements OnInit {

  @Select(AppSelector.sidebarMenu)
  sidebarMenu$: Observable<NbMenuItem[]>;
  @Select(AppSelector.isFetching)
  isFetching$: Observable<boolean>;

  sidebarVisible: boolean = true;
  protected readonly ICONS = ICONS;

  constructor(
    private nbMenuService: NbMenuService,
    private sidebarService: NbSidebarService,
    private userService: UserService,
    private router: Router,
    private store: Store,
    @Inject(NB_WINDOW) private window) {
  }

  ngOnInit() {
    this.nbMenuService.onItemClick()
      .pipe(
        filter(({ tag }) => tag === 'my-context-menu'),
        map(({ item: { title } }) => title),
      )
      .subscribe(title => this.router.navigate(['/authentication']));
  }

  toggleSidebar() {
    this.sidebarService.toggle(this.sidebarVisible);
    this.sidebarVisible = !this.sidebarVisible;
    return false;
  }

  getContextMenu() {
    return this.userService.isLoggedIn() ? [] : NOT_LOGGED_IN_CONTEXT_MENU;
  }

  getImageBySize() {
    return window.screen.width <= 472 ? 'assets/favicon.svg' : 'assets/logo.svg' ;
  }
}
