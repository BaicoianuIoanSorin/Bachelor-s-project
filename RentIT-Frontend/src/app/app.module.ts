import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import {
  NbThemeModule,
  NbLayoutModule,
  NbCardModule,
  NbButtonModule,
  NbPopoverModule,
  NbTabsetModule,
  NbSelectModule,
  NbTooltipModule,
  NbFormFieldModule,
  NbInputModule,
  NbMenuModule,
  NbSearchModule,
  NbListModule, NbSpinnerModule, NbIconModule, NbActionsModule, NbDialogModule, NbToastrModule, NbSidebarModule
} from '@nebular/theme';
import { NbEvaIconsModule } from '@nebular/eva-icons';
import {HttpClientModule} from "@angular/common/http";
import {NgxsModule} from "@ngxs/store";
import {environment} from "../environment/environment.prod";
import {NgxsLoggerPluginModule} from "@ngxs/logger-plugin";
import { ProductsComponent } from './products/products.component';
import { NotFoundComponent } from './not-found/not-found.component';
import {AgGridModule} from "ag-grid-angular";


export const STATES = [

];

export const PAGE_COMPONENTS = [
  AppComponent,
  ProductsComponent,
  NotFoundComponent,
];

export const CORE_COMPONENTS = [

];

export const PROVIDERS = [

];

export const NEBULAR_MODULES = [
  NbSidebarModule.forRoot(),
  NbToastrModule.forRoot(),
  NbLayoutModule,
  NbDialogModule.forRoot(),
  NbButtonModule,
  NbEvaIconsModule,
  NbActionsModule,
  NbMenuModule,
  NbCardModule,
  NbIconModule,
  NbSpinnerModule,
  NbListModule,
  AgGridModule,
  NbSearchModule,
  NbInputModule,
  NbFormFieldModule,
  NbTooltipModule,
  NbSelectModule,
  NbTabsetModule,
  NbPopoverModule,
];

@NgModule({
  declarations: [
   PAGE_COMPONENTS,
    ...CORE_COMPONENTS,
  ],
  imports: [
    BrowserModule,
    HttpClientModule,
    AppRoutingModule,
    BrowserAnimationsModule,
    //NgxsModules setup
    NgxsModule.forRoot(STATES, {
      developmentMode: !environment.production,
    }),
    NgxsLoggerPluginModule.forRoot({disabled: environment.production}),
    NbThemeModule.forRoot({ name: 'corporate' }),
    NEBULAR_MODULES,
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
