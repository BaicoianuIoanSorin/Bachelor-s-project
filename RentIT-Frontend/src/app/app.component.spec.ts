import { Spectator } from "@ngneat/spectator";
import { createComponentFactory } from "@ngneat/spectator/jest";
import { NgxsModule, Store } from "@ngxs/store";
import { RouterTestingModule } from "@angular/router/testing";
import { BrowserModule } from "@angular/platform-browser";
import { BrowserAnimationsModule } from "@angular/platform-browser/animations";
import { HttpClientModule } from "@angular/common/http";
import {NEBULAR_MODULES, PROVIDERS, STATES} from "./app.module";
import {AppRoutingModule} from "./app-routing.module";
import {AppComponent} from "./app.component";

describe('app > app.component.spec.ts', () => {
  let spectator: Spectator<AppComponent>;

  const createComponent = createComponentFactory({
    component: AppComponent,
    imports: [
      RouterTestingModule,
      BrowserModule,
      HttpClientModule,
      AppRoutingModule,
      BrowserAnimationsModule,
      NgxsModule.forRoot(STATES),
      ...NEBULAR_MODULES,
    ],
    providers: [
      ...PROVIDERS,
      Store,
    ],
  });

  beforeEach(() => {
    spectator = createComponent();
  });

  it('should create', () => {
    expect(spectator.component).toBeTruthy();
  });

  it('should match snapshot', async () => {
    spectator.detectChanges();
    await spectator.fixture.whenStable();
    spectator.detectChanges();
    await spectator.fixture.whenRenderingDone();

    // assert
    expect(spectator.fixture).toMatchSnapshot();
  });

  // it('should render title', () => {
  //   spectator.detectChanges();
  //   const compiled = spectator.fixture.nativeElement as HTMLElement;
  //   expect(compiled.querySelector('.content span')?.textContent).toContain('RentIT-Frontend app is running!');
  // });
});
