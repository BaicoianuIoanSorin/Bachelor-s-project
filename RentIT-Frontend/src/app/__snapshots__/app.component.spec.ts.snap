// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app > app.component.spec.ts should match snapshot 1`] = `
<app-root
  title={[Function String]}
>
  App component
</app-root>
`;
